src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -275,15 +289,84 @@ def _get_prioritized_spans(
             span_metadata_size = spans_to_send_sizes[index]
             span_size = get_event_base64_size(span)
 
-            if current_size + span_size - span_metadata_size < request_max_size:
+            if current_size + span_size - span_metadata_size < buffered_max_size:
                 spans_to_send_dict[index] = span
                 current_size += span_size - span_metadata_size
             else:
                 # This is an optimization step. If the spans are too big, don't try to send them.
                 too_big_spans += 1
                 if too_big_spans >= too_big_spans_threshold:
                     break
-    return list(spans_to_send_dict.values())
+
+        # If we dropped spans we need to update the enrichment spans dropped spans reasons
+        final_spans_list = list(spans_to_send_dict.values())
```

Comment:
put this line in the else of them if

Reply:
I didn't do that on purpose, in case the `lumigo_safe_execute` crashes I still want an answer to return (Better return the spans without returning drops then not returning any data)